index.js

```diff
@@ -11,96 +11,185 @@ const Type = (exports.Type = {
-        break;
-      }
-      case Type.MESH: {
-        collisionShape = _createTriMeshShape(sceneRoot, mergeGeometry);
-        break;
+    const createCollisionShape = function(root) {
+      bounds.min.set(0, 0, 0);
+      bounds.max.set(0, 0, 0);
+      localOffset.set(0, 0, 0);
+
+      let collisionShape;
+      switch (type) {
+        case TYPE.BOX: {
+          const hx = fit === FIT.MANUAL ? options.halfExtents : computeHalfExtents(root);
+          collisionShape = _createBoxShape(hx);
+          break;
+        }
+        case TYPE.CYLINDER: {
+          const hx = fit === FIT.MANUAL ? options.halfExtents : computeHalfExtents(root);
+          collisionShape = _createCylinderShape(hx, cylinderAxis);
+          break;
+        }
+        case TYPE.CAPSULE: {
+          const hx = fit === FIT.MANUAL ? options.halfExtents : computeHalfExtents(root);
+          collisionShape = _createCapsuleShape(hx, cylinderAxis);
+          break;
+        }
+        case TYPE.CONE: {
+          const hx = fit === FIT.MANUAL ? options.halfExtents : computeHalfExtents(root);
+          collisionShape = _createConeShape(hx, cylinderAxis);
+          break;
+        }
+        case TYPE.SPHERE: {
+          const radius =
+            fit === FIT.MANUAL && !isNaN(options.sphereRadius) ? options.sphereRadius : computeRadius(root);
+          collisionShape = new Ammo.btSphereShape(radius);
+          collisionShape.sphereRadius = radius;
+          break;
+        }
+        case TYPE.HULL: {
+          const hx = fit === FIT.MANUAL ? options.halfExtents : computeHalfExtents(root);
+          collisionShape = _createHullShape(root, fit, margin, bounds, options.hullMaxVertices || 100000);
+          break;
+        }
+        case TYPE.MESH: {
+          collisionShape = _createTriMeshShape(root, fit);
+          localOffset.add(pos);
+          break;
+        }
+        default:
+          console.warn(type + " is not currently supported");
+          return null;
       }
-      default:
-        console.warn(type + " is not currently supported");
-        return null;
-    }
+      //TODO: Support convex hull decomposition, compound shapes, gimpact (dynamic trimesh)
 
-    collisionShape.type = type;
-    collisionShape.setMargin(margin);
-    collisionShape.destroy = () => {
-      for (let res of collisionShape.resources || []) {
-        Ammo.destroy(res);
+      collisionShape.type = type;
+      collisionShape.setMargin(margin);
+      collisionShape.destroy = () => {
+        for (let res of collisionShape.resources || []) {
+          Ammo.destroy(res);
+        }
+        Ammo.destroy(collisionShape);
+      };
+
+      const localTransform = new Ammo.btTransform();
+      const rotation = new Ammo.btQuaternion();
+      localTransform.setIdentity();
+
+      if (fit === FIT.COMPOUND) {
+        localOffset.add(offset);
+        quat.multiply(orientation);
+      } else {
+        localOffset.copy(offset);
```

Comment:
It's very confusing to think about, that we have this `localOffset` such that:

- We initialize it to zero.
- Then, in the switch statement, if the fit is manual, some branches will compute it based on the bounds, but some branches won't.
- Then, we will use it here.

It's hard for me to tell whether it will have the right value here for every combination of fit and shape.

It would be nice if this were either rearranged to be obviously correct, or there were some comments explaining the "meaning" of stuff like `localOffset` in a way that I could see it was staying correct in every case.